src/os/bluestore/BlueStore.h

```diff
@@ -1553,6 +1563,14 @@ class BlueStore : public ObjectStore,
 
     std::set<OnodeRef> onodes;     ///< these need to be updated/written
     std::set<OnodeRef> modified_objects;  ///< objects we modified (and need a ref)
+
+    // A map from onode to the object offset.  For new objects created in the
+    // transaction the offset is 0, for overwritten objects the offset is the
+    // previous ondisk offset, and for truncated objects the offset is the
+    // negative of the previous ondisk offset.  See update_per_zone_metadata
+    // function for how this map is used.
+    std::map<OnodeRef, int64_t> onode_to_offset_map;
```

Comment:
IIUC TransContext might back multiple operations on the same onode, e.g. a sequence: truncate, write
Hence this map woulldn't be able to support this - the last op is kept only.

Reply:
Thanks!  Fixed it.